rewrite-core/src/main/java/org/openrewrite/tree/ParsingExecutionContextView.java

```diff
@@ -96,16 +95,15 @@ public ParsingExecutionContextView setExcludePattern(Class<? extends Parser> par
     }
 
     public <T extends Parser> boolean accepts(T parser, Path path) {
-        for (String pattern : getExcludePattern(parser.getClass())) {
+        for (String pattern : getExcludePattern(parser.getParserClass())) {
```

Comment:
This new `getParserClass()` is an unfortunate implementation detail that surfaces in the API. I wonder if we can find a cleaner solution. The `Parser.Builder#getDslName()` looks like it would have been nicer, but unfortunately we don't have access to that.

Reply:
I'm providing a better implementation.